src/app/data/typeChart.ts

```diff
@@ -132,23 +146,30 @@ export function calcTypeMatchup(atk: AttackerData, def: DefenderData) {
             atkAbilityCalc = 1.3;
         }
     }
-    const atkMove = atk.move;
-    if (atkMove !== undefined) {
-        const doubleDealtMatch = doubleDealtMoves.find((x) => x.move == atkMove.id);
+
+    let atkMoveCalc = 1;
+    if (atk.move != undefined && !isNull(atk.move) && atk.move.isAttackingMove()) {
```

Comment:
While I admit that semantically isNull seems like it should operate on objects exclusively, it does also already include an undefined check for situations like this. I need to remember that myself, more often.